subcommand/server-acl-init/command.go

```diff
@@ -739,11 +854,85 @@ func (c *Command) configureConnectInject(logger hclog.Logger, consulClient *api.
 		BindName:    "${serviceaccount.name}",
 		Selector:    c.flagBindingRuleSelector,
 	}
-	return c.untilSucceeds(fmt.Sprintf("creating acl binding rule for %s", authMethodTmpl.Name),
-		func() error {
-			_, _, err := consulClient.ACL().BindingRuleCreate(&abr, nil)
+
+	// Add a namespace if appropriate
+	// If namespaces and mirroring are enabled, this is not necessary because
+	// the binding rule will fall back to being created in the Consul `default`
+	// namespace automatically, as is necessary for mirroring.
+	if c.flagEnableNamespaces && !c.flagEnableInjectK8SNSMirroring {
+		abr.Namespace = c.flagConsulInjectDestinationNamespace
+	}
+
+	// If the binding rule already exists and namespaces are enabled, update it
+	if len(existingRules) > 0 && c.flagEnableNamespaces {
```

Comment:
We need to make another call here to populate `existingRules`, this time with the namespace set to `c.flagConsulInjectDestinationNamespace`

Reply:
Updated the logic for the auth method and binding rule checking.